File: javatest/omnia/data/cache/MemoizedTest.java
package omnia.data.cache;

import static org.junit.Assert.assertTrue;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class MemoizedTest {

  @Test
  public void memoize_didReturnSimpleMemoizer() {
    assertTrue(Memoized.memoize(Object::new) instanceof SimpleMemoizer);
  }
}

File: omnia/src/jvmTest/omnia/data/cache/MemoizedTest.kt
package omnia.data.cache

import com.google.common.truth.Truth.assertThat
import kotlin.test.Test
import omnia.data.cache.Memoized.Companion.memoize
import omnia.data.cache.SimpleMemoizer

class MemoizedTest {

  @Test
  fun memoize_didReturnSimpleMemoizer() {
    assertThat(memoize<Any> { Object() }).isInstanceOf(SimpleMemoizer::class.java)
  }
}